src/Algebra/Graph/AdjacencyMap.hs

```diff
@@ -104,7 +104,7 @@ edges = fromAdjacencySets . map (fmap Set.singleton)
 -- 'isEmpty' . overlays == 'all' 'isEmpty'
 -- @
 overlays :: Ord a => [AdjacencyMap a] -> AdjacencyMap a
-overlays = AM . Map.unionsWith Set.union . map adjacencyMap
+overlays = foldr overlay empty
```

Comment:
Wait, not here :) Adjacency maps and relations have a more efficient implementation!

Reply:
I had some doubt, I am going to revert !